File: catowseriOS/CoreBrowser/Tabs/TabsStates.swift
//
//  TabsStates.swift
//  CoreBrowser
//
//  Created by Andrei Ermoshin on 5/30/22.
//  Copyright © 2022 Cotton/Catowser Andrei Ermoshin. All rights reserved.
//

import Foundation
import AutoMockable

public protocol TabsStates: AutoMockable {
    var addPosition: AddedTabPosition { get async }
    var contentState: Tab.ContentType { get async }
    var addSpeed: TabAddSpeed { get }
    var defaultSelectedTabId: UUID { get }
}

File: catowserAndroid/browser-content/src/main/java/org/cotton/browser/content/service/tab/TabsStates.kt
package org.cotton.browser.content.service.tab

import org.cotton.browser.content.data.tab.AddedTabPosition
import org.cotton.browser.content.data.tab.ContentType
import org.cotton.browser.content.data.tab.TabAddSpeed
import java.util.UUID

interface TabsStates {
    val addPosition: AddedTabPosition
    val contentState: ContentType
    val addSpeed: TabAddSpeed
    val defaultSelectedTabId: UUID
}